### PR TITLE
Simply pingpassthrough logic, add fabric ping passthrough

### PR DESCRIPTION
**Pull request by Konicai**
- I wasn't able to reproduce #3757, although I was expecting to be able to by pinging the server while it was starting up (since the pingpassthrough fields in each geyser bootstrap are initialized kind of late). Regardless, `getGeyserPingPassthrough()` cannot be assured to be nonnull, and this should prevent any edge cases.
- Removes unused QueryPacketHandler (since Protocol v3) and anything that became unused (eg protocol passthrough)
- Fabric ping passthrough. Tested with MiniMOTD.

Currently this needs more tested but is ready for review.

### CODE REVIEW

File: core/src/main/java/org/geysermc/geyser/network/netty/GeyserServer.java [hunk 1]
```diff
@@ -142,8 +143,9 @@ private ServerBootstrap createBootstrap(EventLoopGroup group) {
                 if (System.getProperties().contains("disableNativeEventLoop")) {
                     this.geyser.getLogger().debug("EventLoop type is NIO because native event loops are disabled.");
                 } else {
-                    this.geyser.getLogger().debug("Reason for no Epoll: " + Epoll.unavailabilityCause().toString());
-                    this.geyser.getLogger().debug("Reason for no KQueue: " + KQueue.unavailabilityCause().toString());
+                    // Use lambda here, not method reference, or else NoClassDefFoundError for Epoll/KQueue will not be caught
+                    this.geyser.getLogger().debug("Reason for no Epoll: " + throwableOrCaught(() -> Epoll.unavailabilityCause()));
```
Comment (Camotoy) [inline core/src/main/java/org/geysermc/geyser/network/netty/GeyserServer.java:147]: Might want to also gate these statements in a debug mode check just to speed up processing?

Comment (Konicai) [reply core/src/main/java/org/geysermc/geyser/network/netty/GeyserServer.java:147]: It already is, at the very top

**Review by onebeastchris — APPROVED** [review]
Needs a rebase/merge upstream since sponge is now removed, and probably some testing, but code wise, lgtm

**Comment by Konicai** [comment]
only tested on fabric with/without MiniMOTD - haven't tested other platforms in ages

**Pull request merged**